### PR TITLE
create a menu to show a list of all question types and render the select component type

### PR DESCRIPTION
**Pull request by yesyash**
### Note for reviewers
- the folder structure for editable div and related tests and hooks has been changed thus files changed are 33.

### Changes made
- Modify the editable div to support the menu list component
- create a menu list component to show the list of all supported question types

### Test coverage
image

### NOTE:
- in the `editable-div` component the branch which is not checked is `if (!currentElement) return;` which is used when we try to select the current element.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "axios": "^1.6.0",
     "class-variance-authority": "^0.7.0",
     "lucide-react": "^0.277.0",
+    "nanoid": "^5.0.4",
```
Comment (iamitprakash) [inline package.json]: can we remove ^

File: src/modules/form-builder/components/editable-div/editable-div.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    const isFormTitle = value.type === BlockTypesEnum.FORM_TITLE;
+
+    const handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
+        const currentValue = e.currentTarget.innerText;
+        const isPrevKeyShift = prevKey === KeyCodesEnum.SHIFT;
+
+        const isValueEmpty = !currentValue || currentValue?.length === 0 || currentValue === "\n";
+
+        const elements = document.querySelectorAll("[contenteditable]");
```
Comment (iamitprakash) [inline src/modules/form-builder/components/editable-div/editable-div.tsx]: document.querySelectorAll within handleKeyDown can be inefficient, especially if it's called frequently. Consider caching the element collection or using a more efficient approach for finding the current element's index.

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (yesyash) [reply package.json]: - we use it in form builder store to assign id to a block when it is created.

File: src/modules/form-builder/components/editable-div/editable-div.tsx (hunk 2 above)
Comment (yesyash) [reply src/modules/form-builder/components/editable-div/editable-div.tsx]: - removed this, using the index we get when mapping over all the blocks instead.

#### Test coverage post the change:
image

**Review by prakashchoudhary07 — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (iamitprakash) [reply package.json]: please delete this

Comment (Achintya-Chatterjee) [inline package.json:24]: `"nanoid": "5.0.4"`

Comment (Achintya-Chatterjee) [inline package.json]:
```suggestion
    "nanoid": "5.0.4",
```

File: src/modules/form-builder/components/editable-div/editable-div.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        currentElement.innerHTML = "";
+        setCursorToEnd(currentElement);
+        onChange && onChange({ blockId: value.id, payload: { data: "" }, type: blockType });
+    };
+
+    /**
+     * Set initial value to the block
+     * */
+    useEffect(() => {
+        const currentElement = ref.current;
```
Comment (Achintya-Chatterjee) [inline src/modules/form-builder/components/editable-div/editable-div.tsx:97]: please provide a meaningful varible name `currentElement `

File: src/modules/form-builder/components/editable-div/editable-div.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            setBlockInFocus(newBlockPosition);
+        }
+
+        onKeyDown && onKeyDown(e);
+        setPrevKey(e.key);
+    };
+
+    const handleInput = (e: FormEvent<HTMLDivElement>) => {
+        const currentElement = e.currentTarget;
+        const currentValue = currentElement.innerHTML;
```
Comment (Achintya-Chatterjee) [inline src/modules/form-builder/components/editable-div/editable-div.tsx:74]: please  provide a meaning variable name `currentValue`

**Review by iamitprakash — APPROVED** [review]

File: src/modules/form-builder/components/editable-div/editable-div.tsx (hunk 3 above)
Comment (yesyash) [reply src/modules/form-builder/components/editable-div/editable-div.tsx:97]: - here currentElement refers to the current Editable div, do you have any other suggestions for this variable?

File: src/modules/form-builder/components/editable-div/editable-div.tsx (hunk 4 above)
Comment (yesyash) [reply src/modules/form-builder/components/editable-div/editable-div.tsx:74]: - as the value here refers to the current editable-div's value i have named it `currentValue` can you please suggest an alternative?

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**